### DISCUSSION TITLE
How do I fix the pistons?

### DISCUSSION THREAD

**Discussion by isnotimportant** [GriefPrevention/GriefPrevention#1818] [Q&A]
Pistons just don't work anymore and I don't really know how to fix it.

https://user-images.githubusercontent.com/31285332/155864753-f8ddaeca-4aba-4ae4-968b-c67e89f5ad6f.mp4

**Comment by Jikoo** [comment]
It's pretty clear if you bother to read your chat. You have pistons set to not move blocks outside of claims, and the area is not claimed. Claim your stuff or change [piston modes](https://github.com/TechFortress/GriefPrevention/wiki/Setup-and-Configuration#pistonmovement).

**Reply by isnotimportant** [reply]
It's literally in a claim, I also finally found it in the config after rereading it about 5 times.

**Reply by Jikoo** [reply]
> It's literally in a claim

Your video does not display any claim boundaries and GP does not send that message unless the area is unclaimed. It's possible that the area is claimed but the claim bounds do not extend down that far, but since you didn't bother to show claim boundaries in a video displaying a problem, one kinda assumes you didn't bother to claim.

> I also finally found it in the config after rereading it about 5 times

I suggest using your editor's search functionality. Most support the key combination `ctrl+f`. For vim, the search command is `/`. I know this sounds like sarcasm, but seriously, love yourself, search for keywords. Finding anything manually sucks.

If you continue to have issues, please update to a current build of GP, reproduce, and open a ticket including all of the requested information. I recommend running 0e3a2e9 (downloadable [here](https://jitpack.io/com/github/TechFortress/GriefPrevention/0e3a2e9791/GriefPrevention-0e3a2e9791.jar)) instead of 16.18-RC1 due to a fixed bug with automatic claim depth extension.

**Reply by MousePrincess** [reply]
I'm having issues with this too, from what I can tell it has to do with Y level. I have it set to claims_only, on the surface pistons work fine, but If I go down to near bedrock level where someone on the server is based pistons are broken entirely.  The area is claimed- I've checked. When untrusted it says I can't build there.
If you figure out a solution I'd appreciate to know, for now I'm just setting pistons to EVERYWHERE_SIMPLE and hope it works.

**Reply by Jikoo** [reply]
Please make sure you are running an updated build of GP and have edited your configured claim `MaximumDepth` to allow claims to extend down far enough to encompass the pistons. If you still encounter issues, please open an actual ticket with the requested information. It's very hard to piece together what could be wrong with your setup with no details.
